vars/buildPlugin.groovy

```diff
@@ -219,3 +219,21 @@ static List<Map<String, String>> getConfigurations(Map params) {
     }
     return ret
 }
+
+/**
+ * Get recommended configurations for testing.
+ * Includes testing Java 8 and 11 on the newest LTS.
+ */
+static List<Map<String, String>> recommendedConfigurations() {
+    //TODO: replace by 2.164.1 once it is released
+    def recentLTS = "2.164"
+    def configurations = [
+        [ platform: "linux", jdk: "8", jenkins: null ],
+        [ platform: "windows", jdk: "8", jenkins: null ],
+        [ platform: "linux", jdk: "8", jenkins: recentLTS, javaLevel: "8" ],
+        [ platform: "windows", jdk: "8", jenkins: recentLTS, javaLevel: "8" ],
+        [ platform: "linux", jdk: "11", jenkins: recentLTS, javaLevel: "8" ],
+        [ platform: "windows", jdk: "11", jenkins: recentLTS, javaLevel: "8" ]
```

Comment:
Specifying `javaLevel` here seems wrong. It is overriding whatever it is defined in `pom.xml`. Is this just here as a workaround for old parent POMs which do not require an explicit `java.level`? If so, can we make it detect that version; or at least include a TODO comment that this should be reconsidered?

Reply:
Yes, I believe we simply underestimated the speed at which at some point some plugins would start having their `jenkins.version` ≥ the current LTS specified here.